src/main/java/dev/ebyrdeu/movie_jakarta/repository/MovieRepository.java

```diff
@@ -0,0 +1,54 @@
+package dev.ebyrdeu.movie_jakarta.repository;
+
+import dev.ebyrdeu.movie_jakarta.entity.Movie;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.TypedQuery;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+public class MovieRepository {
+    private EntityManager entityManager;
```

Comment:
you need PersistenceContext annotation
```suggestion
    @PersistenceContext(unitName = "mysql")
    private EntityManager entityManager;
```